### PR TITLE
feat(widget-file): allow adding file from url

### LINKED ISSUE #1293: Feature: 'image by URL' editor component
**- Do you want to request a *feature* or report a *bug*?**

Feature

**- What is the current behavior?**

screen shot 2018-04-20 at 14 04 04

**- What is the proposed behavior?**

screen shot 2018-04-20 at 14 05 46

- an alternative to uploading an image to Netlify CMS, by referencing an external image instead.

This is partially possible by registering another editor component (see below), but this new editor component gets in conflict with the existing [image editor component](https://github.com/netlify/netlify-cms/blob/8849953b50c54db69ca078a85a5f614892c22f98/src/components/MarkdownPlugins/image.js#L12) (as it's using the same pattern match). However, this component calls the `toBlock` function correctly when first added into the rich editor block, but after changing context, the text is matched to the existing image component. 

```JS
CMS.registerEditorComponent({
  id: 'image_by_url',
  label: 'Image by URL',
  fields: [
    {name: 'url', label: 'Image URL', widget: 'string'},
    {name: 'alt', label: 'Alternative text', widget: 'string'}
  ],
  pattern: /^!\[(.*)\]\((.*)\)$/,
  fromBlock: function(match) {
    // Netlify CMS doesn't seem to execute this function
    return {
      image: match[2],
      alt: match[1]
    };
  },
  toBlock: function(obj) {
    var url = obj.url || '';
    var alt = obj.alt || '';
    return '![' + alt + '](' + url + ')';
  },
  toPreview: function(obj) {
    // Netlify CMS doesn't seem to execute this function
    var url = obj.url || '';
    var alt = obj.alt || '';
    return '<img src="' + obj.url + '" alt="' + obj.alt + '" />';
  }
});
```

### PR DESCRIPTION
**Pull request by BasixKOR**
Screenshot of two button, one says Chose an image and the other says Insert from URL

**Summary**
Closes #1293, and enble Netlify CMS users to use images from other sources.

**Test plan**
Testing this could be done just like testing with media library, although this one requires the image URL to be reliable. I'm not very sure where to put those tests, so please let me know where should I put it if you think these requires some testing.

**A picture of a cute animal (not mandatory but encouraged)**
My sleeping cat
Here's my cat!

### CODE REVIEW

File: packages/netlify-cms-locales/src/en/index.js [hunk 1]
```diff
@@ -157,11 +157,15 @@ const en = {
       },
       image: {
         choose: 'Choose an image',
+        chooseUrl: 'Insert from URL',
+        promptUrl: 'Please enter URL for your image.',
```
Comment (erezrokah) [inline packages/netlify-cms-locales/src/en/index.js:161]: Can we make this consistent with `linkPrompt`, e.g. `Enter the URL of the image`?

File: packages/netlify-cms-locales/src/en/index.js [hunk 2]
```diff
@@ -157,11 +157,15 @@ const en = {
...
         chooseDifferent: 'Choose different image',
         remove: 'Remove image',
       },
       file: {
         choose: 'Choose a file',
+        chooseUrl: 'Insert from URL',
+        promptUrl: 'Please enter URL for your file.',
```
Comment (erezrokah) [inline packages/netlify-cms-locales/src/en/index.js:168]: Same comment about as the image  `promptUrl`

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @BasixKOR nice work.

Added some comments and also noticed a bug. After inserting a URL you can either replace the image using the media library or remove the existing image. Should we provide an option to replace using a URL?

File: packages/netlify-cms-locales/src/en/index.js (hunk 1 above)
Comment (BasixKOR) [reply packages/netlify-cms-locales/src/en/index.js:161]: Sure! Do you want to change the property name to be consistent with `linkPrompt` too?

**Comment by BasixKOR** [comment]
Providing option to replace with URL is certainly doable, I just forgot that exists lol.

**Review by erezrokah — COMMENTED** [review]
@BasixKOR I followed up with some changes.
If you're good with chose I think this is ready to merge.

**Comment by BasixKOR** [comment]
Oh thank you :) It looks nice to me

**Review by erezrokah — APPROVED** [review]

**Pull request merged**